app/views/licences/tabs/returns.njk

```diff
@@ -43,12 +43,15 @@
       </tr>
     {% endfor %}
     </tbody>
-  {% else %}
-    <p>No returns found</p>
+
+  {% elseif hasReturns == false and hasRequirements == false %}
```

Comment:
Certainly willing to change this if it can be done clearer ?

Reply:
I think we can if we take a different approach. Currently, we are passing through to flags that tell the view whether the licence has returns and has requirements. We are then asking the view to work out what to show.

In my opinion this should be moved to the presenter to try and keep the view as 'dumb' as possible.

The presenter could be changed to work out what the `noReturnsMessage` should be using these flags (means we can then test it easier as well).

The only logic the view then needs is 

```
  {% if noReturnsMessage %}
    // No returns stuff
  {% else %}
    // Display returns table
  {% endif %
```